tomviz/python/Recon_ART.py

```diff
@@ -62,25 +61,21 @@ def transform(self, dataset, Niter=1, Nupdates=0, beta=1.0):
                 break
 
             for s in range(Nslice):
-                if self.canceled:
+                if self.canceled or self.completed:
                     return
-                elif self.completed:
-                    break
 
                 self.progress.message = 'Iteration No.%d/%d,Slice No.%d/%d.' % (
                     i + 1, Niter, s + 1, Nslice) + etcMessage
 
                 #Initialize slice as zeros on first iteration,
                 #or vectorize the slice for the next iter.
-                if (i == 0):
-                    f[:] = 0
-                elif (i != 0):
-                    f[:] = recon[s, :, :].flatten()
+                f[:] = recon[s, :, :].flatten()
 
                 b = tiltSeries[s, :, :].transpose().flatten()
 
                 for j in range(Nrow):
                     row[:] = A[j, :].toarray()
+                    # a = (b[j] - A[j,:].dot(f)) / rowInnerProduct[j]
```

Comment:
If the this line is no longer require, lets just remove it. It will be in version control if you need it in the future.